pkg/helm/helm.go

```diff
@@ -122,6 +128,52 @@ func fetchNewToken(ctx context.Context) (*oauth2.Token, error) {
 	return t, nil
 }
 
+func (h *Hydrator) setDeployNamespace(destDir string) error {
+	if h.DeployNamespace == "" {
+		// do nothing
```

Comment:
This runs after and on top of the hydration step, so if we set the default namespace here, then all resources will get deployed to the default namespace regardless of what namespace they had either before or after hydration, which is not what we want. 

IIUC, what we want is for resources that _do not have any namespace on them_ to be deployed to the default namespace, which is what happens currently and is tested by the test `TestHelmDefaultNamespace` that I added.

I can look into it further to find exactly where, but I think there is logic somewhere later that specifies that resources without a namespace get deployed to the default namespace.

Reply:
It is because the test helm chart (`wordpress`) has a release namespace parameter: `namespace: {{ .Release.Namespace | quote }}`. Thus, when neither `namespace` nor `deployNamespace` is set, it uses the default value of the `namespace`, which is `default`.

We should add a test chart that doesn't have the release namespace parameter, and no `namespace` or `deployNamespace`. I think with your current logic, it should be deployed to the config-management-system.

Reply:
> We should add a test chart that doesn't have the release namespace parameter, and no namespace or deployNamespace. I think with your current logic, it should be deployed to the config-management-system.

`TestHelmDefaultNamespace` does exactly this, it uses this `simple` chart that I added, which contains a deployment ([see it here](https://github.com/GoogleContainerTools/kpt-config-sync/blob/25e9f60c171643a4730639cf58d415ee292acbfc/e2e/testdata/helm-charts/simple/templates/deployment.yaml)) with no release namespace parameter, no `namespace` and no `deployNamespace`.

Reply:
Thank you for checking!